packages/appkit/src/client.ts

```diff
@@ -1242,14 +1243,15 @@ export class AppKit {
 
       this.setUser({ ...(AccountController.state.user || {}), email: user.email })
 
-      const preferredAccountType = (user.preferredAccountType || 'eoa') as W3mFrameTypes.AccountType
+      const preferredAccountType = (user.preferredAccountType ||
+        OptionsController.state.defaultAccountTypes[namespace]) as W3mFrameTypes.AccountType
       this.setPreferredAccountType(preferredAccountType, namespace)
 
       const userAccounts = user.accounts?.map(account =>
         CoreHelperUtil.createAccount(
           namespace,
           account.address,
-          namespace === ConstantsUtil.CHAIN.EVM ? account.type : 'eoa'
+          account.type || OptionsController.state.defaultAccountTypes[namespace]
```

Comment:
Why not using preferredAccountType variable here?

```suggestion
          account.type || preferredAccountType

Reply:
That's this way because here it is mapping the `accounts` from `user`, which seems to may be different from `user.preferredAccountType`